core/status/engine.go

```diff
@@ -132,16 +135,40 @@ func (s *status) checkHealth(ctx echo.Context) error {
 	return ctx.JSON(responseCode, result)
 }
 
-func (s *status) doCheckHealth() core.Health {
+func (s *status) doCheckHealth(ctx context.Context) core.Health {
 	results := make(map[string]core.Health, 0)
+	type checkerResult struct {
+		name    string
+		results map[string]core.Health
+	}
+	// Perform health checks concurrently to speed up the process; some modules query external resources
+	resultChan := make(chan checkerResult)
+	checkCount := 0
 	s.system.VisitEngines(func(engine core.Engine) {
 		checker, ok := engine.(core.HealthCheckable)
-		if ok {
-			for name, result := range checker.CheckHealth() {
-				results[strings.ToLower(checker.Name()+"."+name)] = result
-			}
+		if !ok {
+			// does not support health checking
+			return
 		}
+		checkCount++
+		go func(checker core.HealthCheckable) {
+			subCtx, cancel := context.WithTimeout(ctx, healthCheckTimeout)
```

Comment:
What is the difference between setting a timeout on the http.client which performs a healthcheck, and adding a timeout to the context?

Reply:
In practice nothing, as long as the time-outs are configured the same.

But this way, the caller (health check engine) has control over the time-out. And the API client could cancel the health check by aborting the HTTP request (very niche feature not actually used, probably). But still :-p